server/evaluator.go

```diff
@@ -39,6 +39,43 @@ func (s *Server) Evaluate(ctx context.Context, r *flipt.EvaluationRequest) (*fli
 	return resp, nil
 }
 
+// BatchEvaluate evaluates a request for multiple flags and entities
+func (s *Server) BatchEvaluate(ctx context.Context, r *flipt.BatchEvaluationRequest) (*flipt.BatchEvaluationResponse, error) {
+	s.logger.WithField("request", r).Debug("evaluate")
+	startTime := time.Now()
+
+	// set request ID if not present
+	if r.RequestId == "" {
+		r.RequestId = uuid.Must(uuid.NewV4()).String()
+	}
+
+	resp, err := s.batchEvaluate(ctx, r)
+	if resp != nil {
+		resp.RequestDurationMillis = float64(time.Since(startTime)) / float64(time.Millisecond)
+	}
+
+	if err != nil {
+		return resp, err
+	}
+
+	s.logger.WithField("response", resp).Debug("evaluate")
+	return resp, nil
+}
+
+func (s *Server) batchEvaluate(ctx context.Context, r *flipt.BatchEvaluationRequest) (*flipt.BatchEvaluationResponse, error) {
+	res := flipt.BatchEvaluationResponse{
+		Response: make([]*flipt.EvaluationResponse, 0, len(r.GetFlags())),
+	}
+	for _, flag := range r.GetFlags() {
+		f, err := s.evaluate(ctx, flag)
+		if err != nil {
+			return nil, err
+		}
+		res.Response = append(res.Response, f)
```

Comment:
we'll also need to add the `RequestID` to the response

Reply:
`RequestID` is added on the `BatchEvaluate` method same as `Evaluate` does

Reply:
I see what you meant, fixed it now